virttest/utils_libvirt/libvirt_disk.py

```diff
@@ -119,3 +127,289 @@ def create_custom_metadata_disk(disk_path, disk_format,
+    backing_file_dict = collections.OrderedDict()
+    snapshot_external_disks = []
+    for index in range(len(relative_sub_folders)):
+        key = relative_sub_folders[index]
+        if index == 0 and skip_first_one:
+            continue
+        else:
+            backing_file_dict[key] = "%s.img" % key
+    for key, value in list(backing_file_dict.items()):
+        backing_file_path = os.path.join(root_dir, key)
+        external_snap_shot = "%s/%s" % (backing_file_path, value)
+        snapshot_external_disks.append(external_snap_shot)
+        options = "%s --diskspec %s,file=%s" % (meta_options, disk_target, external_snap_shot)
+        virsh.snapshot_create_as(vm.name, options,
+                                 ignore_status=False,
+                                 debug=True)
+    logging.debug('reuse external snapshots:%s' % snapshot_external_disks)
+    return root_dir, snapshot_external_disks
+
+
+def make_syslink_path_backing_files(pre_set_root_dir, volume_path_list, origin_image_format="qcow2",
+                                    created_image_format="qcow2", syslink_back_chain_lenghth=4):
+    """
+    Create backing chain files of syslink path for one image
+
+    :param pre_set_root_dir: preset root dir
+    :param volume_path_list: volume path list
+    :param origin_image: origin image
+    :param origin_image_format: original image format
+    :param created_image_format: created image format
+    :param syslink_back_chain_lenghth: syslink back chain length
+    :return: absolute path of top active file and syslink back chain files list
+    """
+    root_dir = pre_set_root_dir
+    syslink_folder_list = list(map(chr, range(97, 97 + syslink_back_chain_lenghth)))
+    backing_file_dict = collections.OrderedDict()
+    # Make external relative path backing files.
+    for index in range(len(syslink_folder_list)):
+        key = syslink_folder_list[index]
+        folder_path = os.path.join(root_dir, key)
+        cmd = "mkdir -p %s" % folder_path
+        process.run(cmd, shell=True, ignore_status=False)
+        #Create syslink
+        link_cmd = "ln -s  %s %s" % (volume_path_list[index], os.path.join(folder_path, "%s.img" % key))
+        process.run(link_cmd, shell=True, ignore_status=False)
+        if index == 0:
+            continue
+        else:
+            backing_key = syslink_folder_list[index-1]
+            backing_file_dict[key] = "../%s/%s.img" % (backing_key, backing_key)
+    # Create system link backing files
+    disk_format = origin_image_format
+    for key, value in list(backing_file_dict.items()):
+        backing_file_path = os.path.join(root_dir, key)
+        cmd = ("cd %s && qemu-img create -f %s -o backing_file=%s,backing_fmt=%s %s.img"
+               % (backing_file_path, created_image_format, value, disk_format, key))
+        process.run(cmd, shell=True, ignore_status=False)
+        disk_format = created_image_format
```

Comment:
What's the usage for this line?

Reply:
@kylazhang 
First backchain disk format is passed in by parameter.
After that all subsequent backing disk format need set to created_image_format

Reply:
@kylazhang
First backchain disk format is passed in by parameter.
After that all subsequent backing disk format need set to created_image_format

Reply:
Thanks. And I saw some duplication between this function and make_relative_path_backing_files, could you please write new function for common parts and reduce the duplication?

Reply:
@kylazhang 
wrap those code lines into one separate method:_execute_create_backend_file().